proc_create_data takes "proc_ops" and not "file_operations" now

jeffsauer opened the pull request:
As of linux kernel 5.6, driver would no longer compile when installing with dkms.  Single file change, rtw_proc.c, updating "file_operations" to "proc_ops" fixed.
[rtw_proc.zip](https://github.com/astsam/rtl8812au/files/4456386/rtw_proc.zip)

llccd committed:
Fix kernel hang with active keepalive disabled

kimocoder committed:
Merge pull request #649 from llccd/v5.6.4.2

Fix kernel hang with active keepalive disabled

kimocoder committed:
Merge pull request #644 from 1Jeff1/HW17ACU

Add support for Hawking HW17ACU

kimocoder committed:
Update README.md

CGarces committed:
Fix travis build

vanhoefm committed:
rtl8812a: do not overwrite sequence number of injected frames

The sequence number of inject frames was being overwritten. This prevents
certain older attacks against WEP fragmentation older TKIP attacks. Fix
this by tracking if a frames was injected in monitor mode, and if so, do
not overwrite its sequence number.

The patch also adds a module parameter to revert this behaviour if needed.
By setting rtw_monitor_overwrite_seqnum to 1, sequence numbers will again
be set by the driver/device. This may by useful if user-space relied on
the driver/device to set sequence numbers.

This patch was tested using an Alfa AWUS036ACH.

vanhoefm committed:
rtl8812a: add module parameter to retransmit injected frames

The reliability of some attacks is increased by letting the device
retransmit injected frames. Since it may not always be desired to
retransmit injected frames, add a module parameter to enable it
manually.

This was tested with an Alfa AWUS036ACH. Even when using a spoofed
sender MAC address, the retransmission behaviour is as expected.
That is, when an ACK frame towards the spoofed MAC address is received,
the retransmission will stop.

kimocoder committed:
Merge pull request #661 from vanhoefm/v5.6.4.2

rtl8812a: do not overwrite sequence number of injected frames & control retransmission behaviour

kimocoder committed:
Merge pull request #657 from CGarces/patch-1

Fix travis build

CGarces committed:
Fix compilation error with 5-8-rc1
Remove EOL kernel

kimocoder committed:
Merge pull request #673 from CGarces/kernel_5.9

Fix compilation error with 5-8-rc1

kimocoder committed:
Add checks for p->length and remove some goto statements

kimocoder committed:
Fix connection on resume when CONFIG_ANDROID enabled

kimocoder committed:
Modify makefile to handle empty INSTALL_MOD_PATH

kimocoder committed:
Fix outdated timer_init in BTCOEX routines

kimocoder committed:
Let the linker optimize things as it thinks is best

kimocoder committed:
Allow a little extra size for rtw_hostapd_ioctl() packets

kimocoder committed:
Return zero length output to SIOCGIWESSID when not connected

kimocoder committed:
Fix a possible NUMM dereference

kimocoder committed:
rtl8814a: do not overwrite sequence number of injected frames

kimocoder committed:
Add Travis check for GCC-10

kimocoder committed:
Fix implicit declaration of ‘rtw_resume_process’

kimocoder committed:
Minor cleanup

kimocoder committed:
Fix build when CONFIG_USB_SUSPEND is defined

kimocoder committed:
Remove unused includes

kimocoder committed:
Optimize kernel includes

kimocoder committed:
Fix wrong header guards - found by clang

kimocoder committed:
Fix multiple warnings when using clang

kimocoder committed:
-frename-registers is already in -O3

kimocoder committed:
Get rid of few compiler warnings

kimocoder committed:
Some minor optimizations

kimocoder committed:
Removed 'bc' from dependencies

kimocoder committed:
Update Travis CI to add GCC-10

kimocoder committed:
Implement simple transmit power boost and override

kimocoder committed:
Implement simple transmit power boost and override 2/2

Commit:
Fix a Makefile issue

Roman3349 committed:
Fix cross compilation

Signed-off-by: Roman Ondráček <ondracek.roman@centrum.cz>

Commit:
added device Edimax EW-7611UCB (Realtek RTL8821AU)

kimocoder committed:
Merge pull request #684 from Donneker/v5.6.4.2

added device Edimax EW-7611UCB (Realtek RTL8821AU)

Hurricos committed:
bugfix: Wrap newer construct to allow build on kernel <4.18

kimocoder committed:
Merge pull request #681 from Roman3349/fix-cross-compilation

Fix cross compilation

kimocoder committed:
Merge pull request #689 from Hurricos/v5.6.4.2

bugfix: Wrap newer construct to allow build on kernel <4.18

5kft committed:
bugfix: initialize cfg80211_chan_def structure to zero to fix nl80211_send_chandef kernel trap

kimocoder committed:
Fixed kernel v5.8 support

kimocoder committed:
Update dkms.conf version

kimocoder committed:
Merge pull request #699 from 5kft/chandef-init-fix

bugfix: initialize cfg80211_chan_def structure to zero to fix nl80211…

5kft committed:
fix Kconfig compatibility for kernel v5.9

kimocoder committed:
Merge pull request #713 from 5kft/v5.6.4.2-fix-kconfig

fix Kconfig compatibility for kernel v5.9

kimocoder committed:
bugfix: report 48M not 46.5M

kimocoder committed:
radiotap tx support for injection

kimocoder committed:
Makefile: Simplify DKMS install/remove

kimocoder committed:
Makefile: simplify DKMS install/removal

payano committed:
Makefile: escape grep in DRIVER_VERSION variable

kimocoder committed:
Merge pull request #753 from payano/v5.6.4.2

Makefile: escape grep in DRIVER_VERSION variable

kimocoder committed:
Turned 8814au chipset OFF

kimocoder committed:
Update README.md

kimocoder committed:
Update README.md

kimocoder committed:
Update README.md

kimocoder committed:
Update README.md

pprindeville-wti committed:
issue 766: fix warnings about 'statement with no effect'

pprindeville-wti committed:
issue #768: don't apply ARM options to ARM64

CGarces committed:
Fix kernel 5.10-rc1

brkeh committed:
Update README.md

Use = instead of : when switching mode with modprobe

gaffneyd4 committed:
Fix Raspbian Buster Instructions for v5.6.4.2

Version `v5.6.4.2` does not have a `dkms_install.sh` file, so set export `ARCH` variable in current shell instead.

kimocoder committed:
Merge pull request #773 from CGarces/fix_kernel_5.10

Fix kernel 5.10-rc1

kimocoder committed:
Merge pull request #791 from gaffneyd4/patch-1

Fix Raspbian Buster Instructions for v5.6.4.2

CGarces committed:
Migrate travis CI to github actions

kimocoder committed:
Merge pull request #798 from CGarces/github_actions

Migrate travis CI to github actions

kimocoder committed:
Merge pull request #779 from brkeh/patch-1

Update README.md

kimocoder committed:
Merge pull request #767 from pprindeville/issue#766

issue 766: fix warnings about 'statement with no effect'

kuanyili committed:
README.md: Fix bold text formatting

Fix unclosed HTML bold tag and switch to Markdown format.

kimocoder committed:
Merge pull request #800 from kuanyili/fix-bold-text

README.md: Fix bold text formatting

kimocoder committed:
Merge pull request #769 from pprindeville/issue#768

issue #768: don't apply ARM options to ARM64

lizandroagc commented:
> A partir del kernel 5.6 de Linux, el controlador ya no se compilaría al instalar con dkms. Se corrigió el cambio de archivo único, rtw_proc.c, actualización de "file_operations" a "proc_ops".
> [rtw_proc.zip](https://github.com/astsam/rtl8812au/files/4456386/rtw_proc.zip)
 me darias acceso porfavor

CGarces committed:
Fix memory leak

CGarces committed:
Fix cfg80211_ch_switch_started_notify on kernel 5.11-rc3

See https://github.com/torvalds/linux/commit/669b84134a2be14d333d4f82b65943d467404f87

kimocoder committed:
Merge pull request #811 from CGarces/patch-2

Fix cfg80211_ch_switch_started_notify on kernel 5.11-rc3

kimocoder committed:
Merge pull request #810 from CGarces/fix_kernel_5.10_patch

Fix memory leak

CGarces committed:
Fix error on arm builds

kimocoder committed:
Merge pull request #814 from CGarces/patch-3

Fix error on arm builds

astsam committed:
Add ability to get the current channel by iw utility, based on the patch provided by Ángel <kazimiro69@gmail.com>

astsam committed:
Allow inject management and control frames

astsam committed:
Get per rx path rssi info for CCK modulation

astsam committed:
Add missing 5GHz channels

astsam committed:
Add low frequency 5GHz band

kimocoder committed:
Merge pull request #823 from astsam2/v5.6.4.2

Add ability to get the current channel by iw utility

astsam committed:
Fix error creating an access point below channel 36

CGarces committed:
Improved github actions
Added step to detect misuse of USB_VENDOR_ID_REALTEK
Added check for duplicated USB IDs
Check for .deb files before download it
Move git checkout before build step

CGarces committed:
Remove duplicate USB IDs and sort the elements.

kimocoder committed:
Merge pull request #840 from CGarces/githubactions

Improved github actions

kimocoder committed:
Merge pull request #841 from CGarces/fix_usbids

Remove duplicate USB IDs and sort the elements.

kimocoder committed:
Merge pull request #829 from astsam2/v5.6.4.2

Fix error creating an access point below channel 36

CGarces committed:
Remove rtl8822bu device

CGarces committed:
Fix GRO_DROP deprecation kernel 5.12

kimocoder committed:
Merge pull request #848 from CGarces/fix_5.12

Fix GRO_DROP deprecation kernel 5.12

kimocoder committed:
Merge pull request #847 from CGarces/patch-4

Remove rtl8822bu device

Commit:
Add linux 5.12.x support
Kernel with version >= 5.12.x doesnt support GRO_DROP so the driver must be updated

KonstantIMP committed:
Add linux 5.12.x support

Kernel with version >= 5.12.x doesnt support GRO_DROP so the driver must be updated

CGarces committed:
Fix libc6 dependency 2.33
Fix 18.04 build over default kernel

kimocoder committed:
Merge pull request #860 from CGarces/patch-6

Fix libc6 dependency 2.33

kimocoder committed:
Merge pull request #857 from KonstantIMP/v5.6.4.2

Add linux 5.12 support

CGarces committed:
Fix kernel 5.15 compilation
Remove IPX support from driver, set as obsolete in Jan 2018.
IPX is not supported by the linux kernel since v5.15-rc1 see commit 6c9b40844751ea30c72f7a2f92f4d704bc6b2927

CGarces committed:
fix libc6_2.34 dependency on build

kimocoder committed:
Merge pull request #898 from CGarces/fix_5.15

Fix kernel 5.15 compilation

iyanmv committed:
Remove REMAKE_INITRD

Setting REMAKE_INITRD=no was not needed because it was the default, but since v3.0.0 this option has been deprecated.
https://github.com/dell/dkms/releases/tag/v3.0.0

kimocoder committed:
Add back 8814au support

kimocoder committed:
Merge pull request #913 from iyanmv/iyanmv-remove-remake-initrd

Remove REMAKE_INITRD

kimocoder committed:
Fix kernel v5.15+ compiler warnings

kimocoder committed:
Fixed compiler warning for unused function

kimocoder committed:
Fix (test) for IPX header on various distros

kimocoder committed:
Fix for kernel v3.20

LorenzoAncora committed:
Fix make parallel jobserver warning

LorenzoAncora committed:
Limit the output of "dkms status" and "mkdir"

heitbaum committed:
rtl8812au: Fix code for kernel 5.17 API change

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>

kimocoder committed:
Merge pull request #942 from heitbaum/517

rtl8812au: Fix code for kernel 5.17 API change

kimocoder committed:
Merge pull request #925 from LorenzoAncora/v5.6.4.2

Optimize the output of dkms_install

kimocoder committed:
Optimize kernel v5.15 support

kimocoder committed:
Small code cleanup

Ice-IX committed:
Update Makefile to fix building on ppc

Updated makefile to fix building on ppc32 platforms; note that build will still fail because of an unresolved kernel bug from 2008 (https://bugzilla.kernel.org/show_bug.cgi?id=11143), workaround is to comment out the lines that read "KBUILD_LDFLAGS_MODULE += arch/powerpc/lib/crtsavres.o" in your kernel headers' arch/powerpc/Makefile

Reflexe committed:
linux: set netdev macaddr using dev_addr_set

dev_addr is const now, therefore, we have to use dev_addr_set
in order to modify it.

Reflexe committed:
linux: proc: use pde_data instead of PDE_DATA in >=5.17

Commit:
Update README.md

Add explanation for building on RPi Zero 2 W

CGarces committed:
GCC 11.2 needed for kernel>=5.17. Updated Ubuntu
Removed workaround for libc6 (already on 22.04 LTS).

kimocoder committed:
Fix GCC compiler warning

kimocoder committed:
Merge pull request #963 from CGarces/fix_ci

Partial fix of CI builds

kimocoder committed:
Add kernel 5.18 support

kimocoder committed:
Tag new build version

kimocoder committed:
Merge pull request #962 from 3rd3/patch-2

Update README.md for RPi Zero 2 W

kimocoder committed:
Minor code cleanup

kimocoder committed:
Fix compile warning in rtw_sta_mgt.c

kimocoder committed:
Merge pull request #947 from Ice-IX/patch-1

Update Makefile to fix building on ppc32

kimocoder committed:
Merge pull request #961 from Reflexe/v5.17.5

Fix compilation issues with 5.17 (archlinux)

kimocoder committed:
Add signing to Makefile

kimocoder committed:
Remove more Windows code

kimocoder committed:
Updated README.md

kimocoder committed:
Update README.md

kimocoder committed:
Remove more Windows code

bluegroneman committed:
updated readme flow

Simply moved the git clone section to just before the DKMS build as to improve overall documentation reading experience

kimocoder committed:
Fix compilation for arm64

elfabx committed:
Fix for for corrupted device name

The macro definition for kernels < 5.15 is changed so that it is consistent with usage prior #961. Change should not affect compilation for newer kernels.

DrSchottky committed:
Added RPi Build from CI

heitbaum committed:
Add kernel 6.0 support

heitbaum committed:
Add kernel 5.19.2 support

kimocoder committed:
Merge pull request #992 from heitbaum/kernel-6.0-rc1

Add kernel 5.19.2 and 6.0 support

kimocoder committed:
Merge pull request #989 from elfabx/devicename-fix

Fix for for corrupted device name

kimocoder committed:
Merge pull request #990 from DrSchottky/v5.6.4.2

Added RPi Build from CI

kimocoder committed:
Add some Android fixes

kimocoder committed:
Add switch for VIF (vintf) support

kimocoder committed:
Fix build error for AOSP kernel v5.10

5kft committed:
Add kernel 6.1 support

Co-authored: Rudi Heitbaum <rudi@heitbaum.com>
Signed-off-by: 5kft <5kft@users.noreply.github.com>
Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>

wawrzek committed:
Ensure KVER can be controlled

- allow to build with a kernel other then current on Intel

kimocoder committed:
Merge pull request #1007 from wawrzek/v5.6.4.2

Ensure KVER can be controlled

kimocoder committed:
Fix compiler warnings on kernel 6.x

kimocoder committed:
Merge pull request #1008 from heitbaum/linux-6.1

Add kernel 6.1 support

kimocoder committed:
Merge pull request #978 from bengroneman/v5.6.4.2

updated readme flow

MrR0b0X committed:
Makefile: Fix "fatal error: 'drv_types.h' file not found"

kimocoder committed:
Merge pull request #1029 from MrRob0-X/v5.6.4.2

Makefile: Fix "fatal error: 'drv_types.h' file not found"

MrR0b0X committed:
Makefile: Fix the path of `hal/phydm`

kimocoder committed:
Merge pull request #1030 from MrRob0-X/v5.6.4.2

Makefile: Fix the path of `hal/phydm`

kimocoder committed:
Cleaned up Makefile

MrR0b0X committed:
Makefile: set check for two way path's

this will avoid  error when building standalone or in kernel versions like 5.4

Commit:
added RHEL_RELEASE_VERSION 8

kimocoder committed:
Merge pull request #1032 from MrRob0-X/v5.6.4.2

Makefile: set check for two way path's

GeorgeBannister committed:
fix cfi trip from usb_recv_tasklet

GeorgeBannister committed:
fix cfi trip from rtw_xmit_entry

GeorgeBannister committed:
fix prototype

kimocoder committed:
Merge pull request #1041 from GeorgeBannister/cfi

fix cfi trip from usb_recv_tasklet

kimocoder committed:
Merge pull request #1034 from eusafe/dev

added RHEL_RELEASE_VERSION 8

mjholub committed:
fix(Makefile): ensure akms postinst hooks compatibility

Commit:
Fix control channel frequency in the
 cfg80211_rtw_get_channel() for 40 MHz channels

kimocoder committed:
Add kernel v6.3 support

kimocoder committed:
Merge pull request #1053 from astsam2/v5.6.4.2

Fix control channel frequency in the cfg80211_rtw_get_channel() for 40 MHz channels

kimocoder committed:
Merge pull request #1050 from 154pinkchairs/akms-hooks-compat

Ensure akms postinst hooks compatibility

mjholub committed:
fix(Makefile): remove `[[]]` bashism

kimocoder committed:
Merge pull request #1056 from 154pinkchairs/fix-bashism

fix(Makefile): remove `[[]]` bashism

ferdinandkeller committed:
Add doc for proxmox build instructions

DrSchottky committed:
Fix RPi build github action

Fixed and improved RPi build workflow to support latest driver and Pi kernel versions

gglluukk committed:
Fix compiler (rtw_br_ext.c:969:11) and kernel (net/wireless/sme.c:843) warnings

gglluukk committed:
Adopt patch for 5.x kernel warning dump

kimocoder committed:
Merge pull request #1064 from DrSchottky/v5.6.4.2

Fix RPi build github action

kimocoder committed:
Update GH actions workflow

kimocoder committed:
Revert "Update GH actions workflow"

This reverts commit f6184e3cb0bfc249f0ba624f6661b4bc5c23cc5b.

kimocoder committed:
Merge pull request #1065 from gglluukk/v5.6.4.2

Fix compiler and kernel warnings

kimocoder committed:
Update GH actions workflow

asukiaaa committed:
Add 056E:4010 for elecom device

gglluukk committed:
Pragma GCC < 11 warning fix for -Wstringop-overread

DrSchottky committed:
Parallelize RPi driver builds

kimocoder committed:
Merge pull request #1068 from asukiaaa/patch-1

Add 056E:4010 for elecom WDC-433DU2H2-B

kimocoder committed:
Merge pull request #1069 from gglluukk/v5.6.4.2

Pragma GCC < 11 warning fix for -Wstringop-overread

kimocoder committed:
Possible fix for iwd crash

kimocoder committed:
Update dkms.conf

kimocoder committed:
Fix clang warning

kimocoder committed:
get_ra() exists only on PowerPC

kimocoder committed:
Remove useless warning

kimocoder committed:
Remove dead code associated with CONFIG_PLATFORM_INTEL_BYT

kimocoder committed:
fix SUBARCH on riscv64

kimocoder committed:
Minor fixes and cleanup

kimocoder committed:
Code optimization

DrSchottky committed:
Fixed job name in raspberry action

kimocoder committed:
Some cleanup of unused chips

kimocoder committed:
Merge pull request #1072 from DrSchottky/v5.6.4.2

Parallelize builds on RPi Action

kimocoder committed:
Revert "Some cleanup of unused chips"

This reverts commit b3227c731ec957b706f8eb2ebcf42ba1c2f0d610.

kimocoder committed:
Address warning in createbss_hdl()

CGarces committed:
Use eth_hw_addr_set instead of dev_addr_set

kimocoder committed:
Merge pull request #1089 from CGarces/fix_1088

Use eth_hw_addr_set instead of dev_addr_set

CGarces committed:
Fix compilation error on kernels < 5.15

kimocoder committed:
Merge pull request #1091 from CGarces/patch-1

Fix compilation error on kernels < 5.15

paralin committed:
ioctl_cfg80211: fix build against kernels < 4.10.x

Kernel 4.10.x adjusted cfg80211_connect_bss to accept a timeout as the last
argument. Add a conditional statement which removes this extra parameter if the
kernel is less than version 4.10.x to fix build errors against version 4.9.x.

Note that the Jetson TX2 kernel is still version 4.9.x even though this version
is technically deprecated upstream.

Signed-off-by: Christian Stewart <christian@aperture.us>

kimocoder committed:
Merge pull request #1092 from paralin/fix-kernel-4.9

ioctl_cfg80211: fix build against kernels < 4.10.x

seppzer0 committed:
Add an extra definition for cfg80211_connect_bss() usage

kimocoder committed:
Merge pull request #1097 from seppzer0/v5.6.4.2

ioctl_cfg80211: add an extra definition for cfg80211_connect_bss() calls

kimocoder committed:
Fix the VFS internal message

kimocoder committed:
Add kernel v6.6 support

sarrchri committed:
Fix build on 6.x kernels.

It was broken due to kernel changes backported from 6.5 (torvalds/linux@e8c2af6).
Changes have been tested on the following configurations and observed to build:
 - Debian Bullseye, kernel 6.1.38-4 (bullseye-backports)
 - Debian Bookworm, kernel 6.1.52-1
 - Ubuntu 23.04 Lunar, kernel 6.2.0-36

There are also other kernel versions which are affected up to some point in their history,
but there were a few reasons I ended up not including them:
 - Kernel 6.2: Not in use on Debian, likely skipped in favor of 6.4+. Ubuntu though has backported the change to 6.2 on Lunar.
 - Kernel 6.3: Not in use on Debian/Ubuntu, seems to have been skipped in favor of 6.4/6.5+.
 - Kernel 6.4: Not in use on Ubuntu. Debian Bookworm has this available in backports, but they now also offer 6.5 which does not have the issue.

In case someone tries to build this module running kernel 6.2 on something other than Ubuntu, this fix is most likely necessary.

sarrchri committed:
Fix build on Ubuntu 23.04 (kernel 6.2).

This is done by specifying the ranges for 6.3 and 6.4 versions which need the fix.
Kernel 6.2, to my knowledge, is not used in other popular distros than Ubuntu.

Ubuntu had backported the change from kernel 6.5 to 6.2, which is why this is needed.

Added a comment instructing what to do if one wants to build the module
running kernel 6.2 on something else than Ubuntu.

bwh-mind committed:
ioctl_cfg80211: Add support for standard RSSI notifications

Currently background scanning with wpa_supplicant does not work very
well because rtl8812au does not notify it about changes to RSSI.  To
fix this:

* Implement the cfg80211_ops::set_cqm_rssi_config operation to set the
  parameters for RSSI notifications.
* Add a rtw_cfg80211_cqm_rssi_update() function that calls
  cfg80211_cqm_rssi_notify() if the RSSI has changed significantly
  (based on those parameters).
* When connected in infrastructure mode, call
  rtw_cfg80211_cqm_rssi_update() after processing a beacon and
  updating the RSSI.

Signed-off-by: Ben Hutchings <ben.hutchings@mind.be>

kimocoder committed:
Merge pull request #1119 from sarrchri/fix-build-6x

Fixed build on 6.x kernels.

bwh-mind committed:
Make use of MODULE_IMPORT_NS conditional

Commit 8954f2b8d8a06db44bef9a1318f6e1cdb001b98d added an unconditional:

    MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);

For compatibility with old kernel versions, guard this declaration
with #ifdef MODULE_IMPORT_NS.

This specific symbol namespace is also Android-specific, so the
declaration should also be conditional on building for Android, but
there doesn't seem to be a single preprocessor symbol that indicates
that.

Signed-off-by: Ben Hutchings <ben.hutchings@mind.be>

bwh-mind committed:
Fix eth_hw_addr_set() definition for compatibility with stable branches

eth_hw_addr_set() was added in Linux 5.15 but has now been backported
to 4.19.291, 5.4.251, and 5.10.188.  This currently results in build
failure for these stable branches.

There's no simple way to test for the addition of this function since
LINUX_VERSION_CODE limits version components to 255.

Work around this by defining an inline function rtw_eth_hw_addr_set()
and a macro eth_hw_addr_set().  This effectively shadows any
backported definition of the eth_hw_addr_set() function without any
conflict.

Signed-off-by: Ben Hutchings <ben.hutchings@mind.be>

kimocoder committed:
Merge pull request #1123 from bwh-mind/compat

Kernel compatibility fixes

kimocoder committed:
Merge pull request #1122 from bwh-mind/rssi-notif

ioctl_cfg80211: Add support for standard RSSI notifications

crivasr committed:
Fix change_beacon for kernel 6.7

alium committed:
Update os_intfs.c for kernel 6.8

alium committed:
Update usb_intf.c for kernel 6.8

kimocoder committed:
Merge pull request #1145 from alium/v5.6.4.2

Update for kernel 6.8

kimocoder committed:
Update Makefile

kimocoder committed:
Merge pull request #1134 from crivasr/v5.6.4.2

Fix change_beacon for kernel 6.7

kimocoder committed:
Update Makefile

heitbaum committed:
Update Makefile to error on incompatible-pointer-types

kimocoder committed:
Merge pull request #1147 from heitbaum/patch-2

Update Makefile to error on incompatible-pointer-types

alium committed:
Update ioctl_cfg80211.c -  support for kernel 6.9 

 support for kernel 6.9

kimocoder committed:
Merge pull request #1149 from alium/v5.6.4.2

Update ioctl_cfg80211.c

kimocoder committed:
Remove leftovers of Intel WIDI and take a general cleanup

kimocoder committed:
Add Android 11 docs/scripts to 'android' dir/collection

kimocoder committed:
Add Android 10 docs/scripts to 'android' dir/collection

kimocoder committed:
Add logfile script

virer committed:
Added lastest drv modif from upstream and fix for rhel9.4

chinawrj committed:
Revert "Remove leftovers of Intel WIDI and take a general cleanup"

This reverts commit c7f8f6e3633bd91a9f359eb3406beb972e58cd7b.
This commit is reverted because it introduces such issues as
monitor stops working. Before the issue is solved, we should
keep the code on main branch work

Darin755 committed:
Add note about the issues with AC600

Darin755 committed:
Fix spelling

kimocoder committed:
Update README.md

amazingfate committed:
fix build for kernel v6.13

kimocoder committed:
Merge pull request #1220 from amazingfate/fix-6.13

fix build for kernel v6.13

kimocoder committed:
Merge pull request #1175 from Darin755/v5.6.4.2

Add note about the issues with AC600

kimocoder committed:
Merge pull request #1170 from chinawrj/v5.6.4.2

Revert "Remove leftovers of Intel WIDI and take a general cleanup"

kimocoder committed:
Merge pull request #1062 from ferdinandkeller/v5.6.4.2

Added documentation for proxmox build instructions

kimocoder committed:
Merge pull request #1167 from virer/v5.6.4.2_rocky94

Added support to build on rocky9.4

rpardini committed:
fix build for kernel 6.14

kimocoder committed:
Merge pull request #1226 from rpardini/fix-build-6.14.y

fix build for kernel 6.14

paralin committed:
Fix build for kernel 6.19

With kernel 6.19 and CONFIG_X86_KERNEL_IBT or CONFIG_KLP_BUILD enabled,
objtool runs on the linked module object (delayed objtool via --link).
The module Makefile passes --strip-all to ld -r, which strips local
function symbols. Without these symbols, objtool cannot determine
function boundaries and reports errors like:

  88XXau.o: warning: objtool: rtw_mp_cmd+0xf8: unannotated intra-function call

Replace --strip-all with --strip-debug to preserve function symbols
while still stripping debug information, allowing objtool to properly
validate the linked module object.

Signed-off-by: Christian Stewart <christian@aperture.us>

kimocoder committed:
Merge pull request #1254 from paralin/fix-6.19

Fix build for kernel 6.19
